src/routes/group/chat/[id]/+page.svelte

```diff
@@ -160,18 +160,29 @@
 						<div class="messages">
 							<div class="messages-inner">
 								<!-- Chat bubbles -->
-								{#each messages as message}
+								{#each messages as message, i}
 									{#if message.type === 'user' && message.text.length > 0}
+										{@const len = messages.length}
+										{@const sameSender =
+											messages[i].fromAddress === messages[(i + len - 1) % len].fromAddress}
+										{@const lastMessage =
+											messages[i].fromAddress !== messages[(i + 1) % len].fromAddress ||
+											messages[(i + 1) % len].type !== 'user'}
```

Comment:
@vojtechsimetka 
I am using these to group messages from the same person in a group chat. The first and last messages have the curvy top, the middle ones don't. Also, the name and avatar should only show on the last message of the group -- except for my own messages. Those never show the name or avatar.

I see the issue you're talking about, thank you for flagging it. 
If you have any suggestions, I'm open 😊

Reply:
The modulo is indeed not needed and can cause logical errors. You can simply use `messages[i].fromAddress === messages[i + 1]?.fromAddress` because if you go over the last index then the indexing operator will return `undefined`

Reply:
@agazso 
man that was my first attempt but I forgot the `?` and kept getting undefined errors so I gave up and was trying something else 🤦‍♀️
Thank you!